server/stats.go

```diff
@@ -80,14 +82,22 @@ func (s *Server) GetStats(ctx context.Context, in *pb.GetStatsRequest) (
 			result[dcid] = tmp[dcid].(*pb.ObsTimeSeries)
 		}
 	}
-
 	// Read data from base cache if no branch cache data is found.
 	// This is valid since branch cache is a superset of base cache.
-	if len(result) == 0 {
-		result, err = readStats(ctx, s.btTable, rowList)
-	}
-	if err != nil {
-		return nil, err
+	if len(result) < len(placeDcids) {
+		extraDcids := []string{}
+		for _, dcid := range placeDcids {
+			if _, ok := result[dcid]; !ok {
+				extraDcids = append(extraDcids, dcid)
+			}
+		}
+		extraData, err := readStats(ctx, s.btTable, buildStatsKey(extraDcids, statsVar))
+		if err != nil {
+			return nil, err
+		}
+		for dcid := range extraData {
+			result[dcid] = extraData[dcid]
+		}
 	}
 
 	// Fill missing place data and result result
```

Comment:
replace in.GetPlace() below with "placeDcids" too?

Reply:
Done